### PR TITLE
Eng 741

### PR DESCRIPTION
**Pull request by yardz**

### CODE REVIEW

**Review by traylewin — CHANGES_REQUESTED** [review]
Great job! 
We need a little refactoring and cleanup before its finsihed.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "oreid-js",
-  "version": "3.0.7",
+  "version": "3.0.8-beta.0",
```
Comment (traylewin) [inline package.json:3]: we already have a 3.0.8-beta.0 so bump to 3.0.8-beta.1

File: src/auth/auth.ts [hunk 2]
```diff
@@ -86,6 +89,24 @@ export default class Auth {
       this._localState.saveAccessToken(accessToken)
     }
     this._user = null
+    this.callSubscribers()
+  }
+
+  public subscribe(subscriber: SubscriberAuth) {
```
[the remark this answers is not in the document]
Comment (traylewin) [reply src/auth/auth.ts:95]: Yes I agree that works better. I assume this means making it an Interface

File: src/index.ts [hunk 3]
```diff
@@ -17,6 +19,8 @@ export {
   OreId,
   signCallbackHandler,
   Transaction,
+  User,
```
[the remark this answers is not in the document]
Comment (WarrickFitz) [reply src/index.ts:22]: Show be addressed by the Linter

**Review by WarrickFitz — COMMENTED** [review]
Tray already reviewed. I did not have anything specific to add.

File: src/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (yardz) [reply src/index.ts:23]: I've never seen lint for export, only for imports... But I fixed it.

File: src/auth/auth.ts [hunk 4]
```diff
@@ -111,6 +114,7 @@ export default class Auth {
     // clear accessToken and user
     this._localState.clearAccessToken()
     this._accessTokenHelper.setAccessToken(null)
+    super.callSubscribers()
```
Comment (traylewin) [inline src/auth/auth.ts:117]: Make sure we callSubscribers() wherever we update data we care about in the class....
We are missing it after creating a new User (in get User()). Is there anywhere else missing?

**Review by traylewin — CHANGES_REQUESTED** [review]
Now that this code is reactive (using observables) I realized that we are missing something. The accessToken can expire but the Auth and User wouldn't know that. And the User.isLoggedIn() would report false results since it's using a static accessToken value.
SO I reworked the code to add:
AccessTokenHelper
- Extends Observable
- In setAccessToken() - sets a timer for the time that token expires like which clears accessToken and calls callSubscribers()
Auth
- subscribes to AccessTokenHelper
- when updated, callSubscribers() - which notifies User
User
- when updated, updates _accessToken, callSubscribers() 

All this will notify the React components when the token expires in real time
So user.isLoggedIn will change the moment the accessToken expires
All the code has been committed to this branch.
The easiest way to test this is to update the runAtTime function to expire the accessToken automatically after 5 seconds.

```
  /** Execute a callback function at a given Unix Epoch time */
  static runAtTime(callback: Function, executionEoochTime: number) {
    const now = Date.now()
    console.log('now:', now)
    executionEoochTime = now + 5000
    console.log('executionEoochTime:', executionEoochTime)
    const runAtTime = Math.max(executionEoochTime, now) // if executionEoochTime is in the past, use the current time to run the function immediately
    return setTimeout(callback, runAtTime - now)
  }
```

**Comment by yardz** [comment]
I just tested it and it worked. Can I merge?

@traylewin @WarrickFitz @deckb

**Pull request merged**